### PR TITLE
fix: Add PasswordField to mask input value of the field with `format:…

### PR DESCRIPTION
**Pull request by igarashitm**
… password`

Fixes: #285
Fixes: #503

Screenshot from 2024-05-02 21-14-45
Screenshot from 2024-05-02 21-17-34

### CODE REVIEW

File: packages/ui/src/components/Form/customField/PasswordField.tsx [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+const PasswordFieldComponent = ({ onChange, ...props }: PasswordFieldProps) => {
+  const generatePassword = () => {
+    const length = 12;
+    const charset = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@%()_-=+';
+    let retVal = '';
+    for (let i = 0, n = charset.length; i < length; ++i) {
+      retVal += charset.charAt(Math.floor(Math.random() * n));
+    }
+    return retVal;
+  };
```
Comment (lordrip) [inline packages/ui/src/components/Form/customField/PasswordField.tsx:45]: Could be possible for this function to be moved to the utils folder?

File: packages/ui/src/components/Form/customField/PasswordField.tsx [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+                aria-label="Generate a new suggested password"
+              />
+            }
+          >
+            Use suggested password: <b>{`${generatedPassword}`}</b>
+          </MenuItem>
+        </MenuList>
+      </MenuContent>
+    </Menu>
+  );
```
Comment (lordrip) [inline packages/ui/src/components/Form/customField/PasswordField.tsx:197]: Considering that we need to provide existing passwords and that we're not creating new user accounts through Kaoto (_at least not at the moment_) I wonder if we would be better off without the password generator bit.

**Review by lordrip — COMMENTED** [review]
Looks good @igarashitm, apart from the couple comments, there's something that I'm not quite sure why happens.

At the moment of writing, this is how the form looks like
image

But with this PR, for some reason, those fields get prefilled with autocompletion coming from the browser
image

And more important, those values gets serialized in the YAML 😕

File: packages/ui/src/components/Form/customField/PasswordField.tsx (hunk 2 above)
Comment (igarashitm) [reply packages/ui/src/components/Form/customField/PasswordField.tsx:197]: TBH I have no idea from what background we wanted to introduce password generator, @lhein could you add some context?

**Comment by igarashitm** [comment]
w/r/t the browser auto fill, I know chrome is eager like crazy to fill the username/password thing everywhere, I think it's done if the input name attribute is same on the combination of one `type=text` and one `type=password`

**Comment by lordrip** [comment]
> w/r/t the browser auto fill, I know chrome is eager like crazy to fill the username/password thing everywhere, I think it's done if the input name attribute is same on the combination of one `type=text` and one `type=password`

Maybe this will help: https://developer.mozilla.org/en-US/docs/Web/Security/Securing_your_site/Turning_off_form_autocompletion

File: packages/ui/src/components/Form/customField/PasswordField.tsx (hunk 2 above)
Comment (lhein) [reply packages/ui/src/components/Form/customField/PasswordField.tsx:197]: The idea was that this component provides the visibility switch and the masking functionality ootb. If you can come up with something similar I am not opposed to hear your solution :)

Comment (igarashitm) [reply packages/ui/src/components/Form/customField/PasswordField.tsx:197]: OK so no need to use PF password generator at all, visibility toggle is just a plain button and the masking is HTML standard. I'll remove the password generator.

File: packages/ui/src/components/Form/customField/PasswordField.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+              {!props.error ? props.helperText : props.errorMessage}
+            </HelperTextItem>
+          </HelperText>
+        </FormHelperText>
+      </InputGroupItem>
+      <InputGroupItem>
+        <Button
+          data-testid={'password-show-hide-button'}
+          variant="control"
+          onClick={() => setPasswordHidden(!passwordHidden)}
```
Comment (lordrip) [inline packages/ui/src/components/Form/customField/PasswordField.tsx:62]:
```suggestion
          onClick={() => {
            setPasswordHidden(!passwordHidden);
          }
```

File: packages/ui/src/components/Form/customField/PasswordField.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    props,
+    <InputGroup>
+      <InputGroupItem isFill>
+        <TextInput
+          aria-label={'uniforms text field'}
+          data-testid={'password-field'}
+          name={props.name}
+          isDisabled={props.disabled}
+          validated={props.error ? 'error' : 'default'}
+          onChange={(_event, value) => onChange(value)}
```
Comment (lordrip) [inline packages/ui/src/components/Form/customField/PasswordField.tsx:43]:
```suggestion
          onChange={(_event, value) => {
            onChange(value);
          }
```

File: packages/ui/src/components/Form/customField/PasswordField.tsx [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+const PasswordFieldComponent = ({ onChange, ...props }: PasswordFieldProps) => {
+  const [passwordHidden, setPasswordHidden] = useState<boolean>(true);
+
+  return wrapField(
+    props,
+    <InputGroup>
+      <InputGroupItem isFill>
+        <TextInput
+          aria-label={'uniforms text field'}
+          data-testid={'password-field'}
```
Comment (lordrip) [inline packages/ui/src/components/Form/customField/PasswordField.tsx:39]:
```suggestion
          aria-label="uniforms text field"
          data-testid="password-field"
```

**Review by lordrip — APPROVED** [review]

**Pull request merged**